core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1348,6 +1348,28 @@ public static void writeOperationNoSize(BufferedChecksumStreamOutput out, Transl
     }
 
     /**
+     * Gets the minimum generation that could contain any sequence number after the specified sequence number, or the current generation if
+     * there is no generation that could any such sequence number.
+     *
+     * @param seqNo the sequence number
+     * @return the minimum generation for the sequence number
+     */
+    public TranslogGeneration getMinGenerationForSeqNo(final long seqNo) {
+        try (ReleasableLock ignored = writeLock.acquire()) {
+            final long minTranslogFileGeneration = readers
```

Comment:
I wonder if we can just use a for loop and `Math.min` this is much harder for me to read instead of:
```Java
long minTranslogFileGeneration = this.currentFileGeneration;
for (Reader r : readers) {
  if (seqNo <= r.getCheckpoint().maxSeqId) {
     minTranslogFileGeneration = Math.min(minTranslogFileGeneration, r. getGeneration());
  }
}
```